src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/create/action/CreateSubFlowsAction.java

```diff
@@ -42,11 +43,13 @@ public void perform(State from, State to, Event event, YFlowCreateContext contex
         String yFlowId = stateMachine.getYFlowId();
         Collection<RequestedFlow> requestedFlows =
                 YFlowRequestMapper.INSTANCE.toRequestedFlows(stateMachine.getTargetFlow());
+        stateMachine.setRequestedFlows(requestedFlows);
         log.debug("Start creating {} sub-flows for y-flow {}", requestedFlows.size(), yFlowId);
         stateMachine.clearCreatingSubFlows();
 
-        requestedFlows.forEach(requestedFlow -> {
+        Optional.ofNullable(requestedFlows.iterator().next()).ifPresent(requestedFlow -> {
```

Comment:
Calling next() without checking for hasNext() is the way to get NoSuchElementException, not a null value.

Reply:
Now `.stream().findFirst()` is used.